ch04/README.md

```diff
@@ -362,7 +362,7 @@ int i; double d; const string *ps; char *pc; void *pv;
 pv = (void*)ps; // pv = const_cast<string*>(ps); or pv = static_cast<void*>(const_cast<string*>(ps));
 i = int(*pc);   // i = static_cast<int>(*pc);
 pv = &d;        // pv = static_cast<void*>(&d);
-pc = (char*)pv; // pc = reinterpret_cast<char*>(pv); 
+pc = (char*)pv; // pc = static_cast<char*>(pv);   ref: Page 163.
 ```
```

Comment:
I can't agree with you here.

Page 163 give a example:

```cpp
void* p = &d;
// ok, converts void* back to the original pointer type
double *dp = static_cast<double*>(p);
```

Take notice at the comment: **back to the original pointer type**. Here, `p`'s original pointer type is `double*`. Thus, using `static_cast` is ok.

But in this case, `pv`'s original pointer type is `double` (`pv = &d;`), and we try to convert it to `char*`. It isn't `back to` action. So `reinterpret_cast` is more suitable.